comp110-worksheet-7/DirectoryUtils.cs

```diff
@@ -24,37 +34,70 @@ public static bool IsDirectory(string path)
 		// Return the total size, in bytes, of all the files below the given directory
 		public static long GetTotalSize(string directory)
 		{
-			throw new NotImplementedException();
+			valuesReference = new ImportantValues();
+			void sizeRead() //reads the size of each file in a directory
+			{
+				string[] files = new string[0];
+				files = Directory.GetFiles(directory); //Assigns each file in the directory to an array
+				foreach (string file in files)
+				{
+					valuesReference.fileSize += new FileInfo(file).Length;
+
+					valuesReference.fileDirectories[0] = valuesReference.fileDirectories[0] + file;
+					valuesReference.allSizes[0] = valuesReference.allSizes[0] + valuesReference.fileSize;
+					valuesReference.fileNo = valuesReference.fileNo + 1;
+					if(IsDirectory(file) != true)
+					{
+						valuesReference.fileDepth = valuesReference.fileDepth + 1;
+						Recursion(files[0], file);
+					}
+				}
+			}
+			void Recursion(string files, string file)
+
+			{
+				foreach (var elem in files)
+					valuesReference.fileSize += GetFileSize(file);
+					if(IsDirectory(file))
+					{
+						sizeRead();
+					}
+			}
+			return valuesReference.fileSize;
 		}
 
 		// Return the number of files (not counting directories) below the given directory
 		public static int CountFiles(string directory)
 		{
-			throw new NotImplementedException();
+			return valuesReference.fileNo;
```

Comment:
You're assuming that `GetTotalSize` is the first thing to be called, so that `valuesReference` will be populated -- this assumption might not be true